presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -382,6 +396,76 @@ public synchronized void setOutputBuffers(OutputBuffers newOutputBuffers)
         }
     }
 
+    @Override
+    public ListenableFuture<?> removeRemoteSource(TaskId remoteSourceTaskId)
+    {
+        URI remoteSourceUri = uriBuilderFrom(taskLocation)
+                .appendPath("remote-source")
+                .appendPath(remoteSourceTaskId.toString())
+                .build();
+
+        Request request = prepareDelete()
+                .setUri(remoteSourceUri)
+                .build();
+        RequestErrorTracker errorTracker = new RequestErrorTracker(
```

Comment:
Do we need to create tracker here or maybe we should create this tracker in `removeRemoteSourceHelper` given the only usage is to pass this variable inside?

Reply:
@shixuan-fan : Remember `removeRemoteSourceHelper` recursively calls itself, so we don't want to create `RequestErrorTracker` inside it :)

Reply:
Cool. Didn't see that.